pkg/deploy/dev-workspace/dev_workspace.go

```diff
@@ -62,19 +54,22 @@ func NewDevWorkspaceReconciler() *DevWorkspaceReconciler {
 }
 
 func (d *DevWorkspaceReconciler) Reconcile(ctx *chetypes.DeployContext) (reconcile.Result, bool, error) {
-	if isDevWorkspaceOperatorCSVExists(ctx) {
-		// Do nothing if DevWorkspace has been already deployed via OLM
-		return reconcile.Result{}, true, nil
-	}
-
 	if infrastructure.IsOpenShift() {
-		wtoInstalled, err := isWebTerminalSubscriptionExist(ctx)
-		if err != nil {
+		// Do nothing if explicitly declared not to manage Dev Workspace resources
+		isNoOptDWO, err := isNoOptDWO()
+		if isNoOptDWO {
+			return reconcile.Result{}, true, nil
+		} else if err != nil {
 			return reconcile.Result{Requeue: true}, false, err
 		}
-		if wtoInstalled {
-			// Do nothing if WTO exists since it should bring or embeds DWO
+
+		// Do nothing if Dev Workspace operator has owner (installed via OLM).
+		// In this case Dev Workspace operator resources mustn't be managed by Che operator.
+		isDevWorkspaceOperatorHasOwner, err := isDevWorkspaceOperatorHasOwner(ctx)
+		if isDevWorkspaceOperatorHasOwner {
 			return reconcile.Result{}, true, nil
+		} else if err != nil {
+			return reconcile.Result{Requeue: true}, false, err
 		}
```

Comment:
Do we really care now about the owner?

Reply:
We need this for the following case:
1. OpenShift cluster with preinstalled DevWorkspace operator from OperatorHub
2. Eclipse Che operator is deployed via yaml templates (for instance for testing purpose), so `NO_OPT_DWO=false`

so, che operator must not managed DWO resources here.

Reply:
1. OpenShift cluster with preinstalled DevWorkspace operator from OperatorHub

but in this case, it will still be managed by olm, no? My understanding is that olm should discover this instance during the Eclipse Che / Devspaces installation

2. Eclipse Che operator is deployed via yaml templates (for instance for testing purpose), so NO_OPT_DWO=false

This is actually an interesting part, In this case wouldn't olm try to install it as a dependency again?

Reply:
1. Yes DWO will be still managed by olm. 
2. Since che operator is installed via yaml templates, there is no olm bundle and there is no any dependencies.